src/main/kotlin/com/grepp/nbe1_3_team04/member/service/MemberServiceImpl.kt

```diff
@@ -0,0 +1,121 @@
+
+        memberRepository.save(member)
+        return MemberResponse.from(member)
+    }
+
+    @Transactional
+    override fun login(serviceRequest: LoginServiceRequest): TokenResponse {
+        val member: Member = getMemberByEmail(serviceRequest.email)
+        checkPasswordMatch(serviceRequest.password, member.password)
+
+        val tokenResponse: TokenResponse = jwtTokenUtil.createToken(member.email)
+
+        // Redis에 RefreshToken 저장
+        setRedis(member.email, tokenResponse.refreshToken, tokenResponse.refreshTokenExpirationTime, TimeUnit.MICROSECONDS)
+
+        return tokenResponse
+    }
+
+    @Transactional
+    override fun logout(request: HttpServletRequest): String {
+        val accessToken: String? = jwtTokenUtil.resolveToken(request)
+        val email: String = jwtTokenUtil.getEmailFromToken(accessToken)
+        jwtTokenUtil.tokenValidation(accessToken)
+
+        redisTemplate.opsForValue()[email]?.let {
+            redisTemplate.delete(email)
+        }
+
+        val expiration: Long = jwtTokenUtil.getExpiration(accessToken!!) // tokenValidation 을 통해 accessToken 검증 완료됨
+        setRedis(accessToken, "logout", expiration, TimeUnit.MICROSECONDS)
+
+        return "Success Logout"
+    }
+
+    override fun reissue(request: HttpServletRequest, refreshToken: String?): TokenResponse {
+        var token: String? = refreshToken
+        if (token.isNullOrBlank()) {
+            token = JwtTokenFilter.getRefreshTokenByRequest(request) // 헤더에 없을 경우 쿠키에서 꺼내 씀
+        }
+        jwtTokenUtil.tokenValidation(token)
+
+        return convertToTokenResponseFrom(token!!)
+    }
+
+    @Transactional
+    override fun update(member: Member, request: UpdateServiceRequest): MemberResponse {
+        member.update(request.name, request.phoneNumber, request.gender)
+        memberRepository.save(member)
+
+        return MemberResponse.from(member)
+    }
+
+    @Transactional
+    override fun updatePassword(member: Member, serviceRequest: UpdatePasswordServiceRequest): MemberResponse {
+        checkPasswordMatch(serviceRequest.prePassword, member.password)
+        member.changePassword(passwordEncoder.encode(serviceRequest.newPassword))
+        memberRepository.save(member)
+
+        return MemberResponse.from(member)
+    }
+
+
+    private fun getMemberByEmail(email: String): Member {
+        return memberRepository.findByEmail(email) ?: throw IllegalArgumentException("존재하지 않는 사용자 입니다.")
+    }
+
+    private fun setRedis(key: String, value: String, expirationTime: Long, timeUnit: TimeUnit){
+        redisTemplate.opsForValue().set(key, value, expirationTime, timeUnit)
+    }
+
+    private fun checkDuplicateEmail(email: String){
+        require(!memberRepository.existByEmail(email)) { "이미 존재하는 이메일 입니다." }
+    }
+
+    private fun checkPasswordMatch(password: String?, passwordConfirm: String?){
+        require(jwtSecurityConfig.passwordEncoder().matches(password, passwordConfirm)) { "패스워드가 일치하지 않습니다." }
```

Comment:
혹시 이미 생성자에서 주입받은 passwordEncoder를 사용하지 않는 특별한 이유가 있나요?
```suggestion
        require(passwordEncoder.matches(password, passwordConfirm)) { "패스워드가 일치하지 않습니다." }
```
이렇게 작성했을 때 문제가 있나요?

Reply:
아 맞네요! 같은 곳에서 등록된 빈이라 그냥 passwordEncoder만 쓰는게 좋을 거 같습니다! 수정하겠습니다.